x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -502,57 +609,27 @@ export class ManifestManager {
   public async tryDispatch(manifest: Manifest): Promise<Error[]> {
     const errors: Error[] = [];
 
+    const allPackagePolicies: PackagePolicy[] = [];
     await iterateAllListItems(
       (page) => this.listEndpointPolicies(page),
+      (packagePolicy) => {
+        allPackagePolicies.push(packagePolicy);
+      }
+    );
+
+    await pMap(
+      allPackagePolicies,
       async (packagePolicy) => {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        const { id, revision, updated_at, updated_by, ...newPackagePolicy } = packagePolicy;
-        if (newPackagePolicy.inputs.length > 0 && newPackagePolicy.inputs[0].config !== undefined) {
-          const oldManifest = newPackagePolicy.inputs[0].config.artifact_manifest ?? {
-            value: {},
-          };
-
-          const newManifestVersion = manifest.getSemanticVersion();
-          if (semver.gt(newManifestVersion, oldManifest.value.manifest_version)) {
-            const serializedManifest = manifest.toPackagePolicyManifest(packagePolicy.id);
-
-            if (!manifestDispatchSchema.is(serializedManifest)) {
-              errors.push(
-                new EndpointError(
-                  `Invalid manifest for policy ${packagePolicy.id}`,
-                  serializedManifest
-                )
-              );
-            } else if (!manifestsEqual(serializedManifest, oldManifest.value)) {
-              newPackagePolicy.inputs[0].config.artifact_manifest = { value: serializedManifest };
-
-              try {
-                await this.packagePolicyService.update(
-                  this.savedObjectsClient,
-                  // @ts-expect-error TS2345
-                  undefined,
-                  id,
-                  newPackagePolicy
-                );
-                this.logger.debug(
-                  `Updated package policy ${id} with manifest version ${manifest.getSemanticVersion()}`
-                );
-              } catch (err) {
-                errors.push(err);
-              }
-            } else {
-              this.logger.debug(
-                `No change in manifest content for package policy: ${id}. Staying on old version`
-              );
-            }
-          } else {
-            this.logger.debug(`No change in manifest version for package policy: ${id}`);
-          }
-        } else {
-          errors.push(
-            new EndpointError(`Package Policy ${id} has no 'inputs[0].config'`, newPackagePolicy)
-          );
+        const error = await this.updatePackagePolicyIfNeeded(manifest, packagePolicy);
```

Comment:
Is this a good place to also use the `.bulk()` api to send the updates for the policies?  If I read this correctly, its doing updates one at a time

Reply:
https://github.com/elastic/kibana/pull/160387/commits/74bca2bf0ddb0f15d20e133dfe526f8e3e49df1b